sqlglot/dialects/tsql.py

```diff
@@ -681,6 +681,28 @@ def _parse_unique(self) -> exp.UniqueColumnConstraint:
 
             return self.expression(exp.UniqueColumnConstraint, this=this)
 
+        def _parse_partition(self) -> t.Optional[exp.Partition]:
+            if not self._match_pair(TokenType.WITH, TokenType.L_PAREN) or not self._match_text_seq(
+                "PARTITIONS"
+            ):
```

Comment:
Even though token matching should be faster, we should probably use a single `_match_text_seq` call here to avoid consuming `WITH (` in case we don't find a subsequent `PARTITIONS` token.
```suggestion
            if not self._match_text_seq("WITH", "(", "PARTITIONS"):
```